### DISCUSSION TITLE
About wayland protocal

### DISCUSSION THREAD

**Discussion by Decodetalkers** [rust-windowing/winit#2922] [Q&A] [answered]
I have already seem there is a pr about layershell, but now I have another question

if some day there is a new protocal , like the ext-session-lock-v1, or I want to let winit dispatch virtual-keyboard, then how to handle it in winit? About virtual-keyboard , I maybe will open a new thread to handle it ,but what like ext-session-lock? Seems now to use new protocal I need to make pr to winit, emm

**Comment by kchibisov** [comment, marked as answer]
It's expected that for really specialized clients like that you'll use https://github.com/smithay/client-toolkit directly. I'm not even sure entirely whether we'd need layer shell in the tree.

The reasoning for that is that you'll use very special cased protocols, and I'm not sure winit should do anything about them, since they are exclusive to the used platforms.
